backend/api/Services/AssetDeckService.cs

```diff
@@ -49,20 +57,58 @@ private IQueryable<AssetDeck> GetAssetDecks()
                 .FirstOrDefaultAsync(assetDeck => assetDeck.Id.Equals(id));
         }
 
-        public async Task<AssetDeck> Create(CreateAssetDeckQuery newAssetDeck)
+        public async Task<IEnumerable<AssetDeck>> ReadByAsset(string asset)
+        {
+            return await _context.AssetDecks.Where(a =>
+                a.AssetCode.Equals(asset)).Include(a => a.SafePositions).ToListAsync();
+        }
+
+        public async Task<AssetDeck?> ReadByAssetAndDeck(string asset, string deck)
+        {
+            return await _context.AssetDecks.Where(a =>
+                a.AssetCode.Equals(asset) &&
+                a.DeckName.Equals(deck)
```

Comment:
Asset code and deck name string comparison should be case insensitive

Reply:
Agreed